### DISCUSSION TITLE
IcebergTableMaintainer expireSnapshots keep one snapshot

### DISCUSSION THREAD

**Discussion by lurnagao-dahua** [apache/amoro#2774] [General]
Hi, expireSnapshots iceberg table in IcebergTableMaintainer.
`table
        .expireSnapshots()
        .retainLast(1)
        .expireOlderThan(olderThan)`
Why only keep one snapshot and not support configurable？

**Comment by zhoujinsong** [comment]
Hi,

Method `retainLast` helps to keep at least N snapshots even if they are older than the expiration timestamp[1].
Amoro lets users configure the expiration time by table property:`change.data.ttl.minutes`[2].

I'm not sure if you're asking want to support retaining more snapshots when expiration time is configured.

references:
[1] https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/ExpireSnapshots.java#L72
[2] https://amoro.apache.org/docs/latest/configurations/#data-cleaning-configurations

**Reply by lurnagao-dahua** [reply]
Hi,I want to support retaining more snapshots when expiration time is configured.

**Reply by zhoujinsong** [reply]
Could you provide some specific scenarios to help us understand why you need additional configurations?

**Reply by lurnagao-dahua** [reply]
Iceberg parameter `history.expire.min-snapshots-to-keep` ,default min number of snapshots to keep on the table and all of its branches while expiring snapshots.
We may use this parameter to allow the user to determine the number of snapshots to be retained
